DocumentsFromSnapshotMigration/src/test/java/com/rfs/FullTest.java

```diff
@@ -308,6 +314,90 @@ private DocumentsRunner.CompletionStatus migrateDocumentsWithOneWorker(SourceRep
         }
     }
 
+    public static Stream<Arguments> makeProcessExitArgs() {
+        var sourceImageArgs = SOURCE_IMAGES.stream().map(name -> makeParamsForBase(name)).collect(Collectors.toList());
+        var targetImageNames = TARGET_IMAGES.stream().map(SearchClusterContainer.Version::getImageName).collect(Collectors.toList());
+
+        return sourceImageArgs.stream()
+            .flatMap(a->
+            targetImageNames.stream().map(b->Arguments.of(a[0], a[1], a[2], b)));
+    }
+
+    @ParameterizedTest
+    @MethodSource("makeProcessExitArgs")
+    public void testProcessExitsAsExpected(SearchClusterContainer.Version baseSourceImageVersion,
+            String generatorImage, String[] generatorArgs,
+            String targetImageName) throws Exception {
+
+        try (var esSourceContainer = new PreloadedSearchClusterContainer(baseSourceImageVersion,
+                    SOURCE_SERVER_ALIAS, generatorImage, generatorArgs);
+                OpensearchContainer<?> osTargetContainer =
```

Comment:
Setting up a toxiproxy in between your future process and your target will let you inject some severe timeouts that should cause timeouts to propagate.  Those should eventually cause a lease expiration, which would be really useful to check.

Take a look at KafkaConfigurationCaptureProxyTest for an example of where we use the toxiproxy test container.  You might also be able to route a test to a dummy endpoint that returns 404 (see SimpleNettyHttpServer).  That should cause retries to go on & on.  For testing purposes, you might need to parameterize (at the command line) the initial lease time for either of these.

Reply:
Will investigate; thanks.

Reply:
I did something a bit simpler this time around, but I'd be interested in following up later for more sophisticated testing of the work coordination.